api/test/unit/utils/test_photon.py

```diff
@@ -216,76 +218,133 @@ def raise_exc(*args, **kwargs):
     yield do
 
 
-def test_get_timeout_no_existing_cache_key(
-    capture_exception, setup_requests_get_exception, redis
-):
-    exc = requests.ReadTimeout()
-    setup_requests_get_exception(exc)
+@pook.on
+def test_get_successful_records_response_code(mock_image_data, redis):
+    (
+        pook.get(PHOTON_URL_FOR_TEST_IMAGE)
+        .params(
+            {
+                "w": settings.THUMBNAIL_WIDTH_PX,
+                "quality": settings.THUMBNAIL_QUALITY,
+            }
+        )
+        .header("User-Agent", UA_HEADER)
+        .header("Accept", "image/*")
+        .reply(200)
+        .body(MOCK_BODY)
+        .mock
+    )
 
-    with pytest.raises(UpstreamThumbnailException):
-        photon_get(TEST_IMAGE_URL)
+    photon_get(TEST_IMAGE_URL)
+    month = get_monthly_timestamp()
+    assert redis.get(f"thumbnail_response_code:{month}:200") == b"1"
+    assert (
+        redis.get(
+            f"thumbnail_response_code_by_domain:subdomain.example.com:{month}:200"
+        )
+        == b"1"
+    )
 
-    capture_exception.assert_called_once_with(exc)
 
-    key = f"{settings.THUMBNAIL_TIMEOUT_PREFIX}subdomain.example.com"
-    assert redis.get(key) == b"1"
+alert_count_params = pytest.mark.parametrize(
+    "count_start, should_alert",
+    [
+        (0, True),
+        (1, True),
+        (99, True),
+        (100, False),
+        (999, True),
+        (1000, False),
```

Comment:
Please also include a few tests for values between 1k and 2k, over 2k to count the actual number of alerts emitted.

Reply:
I can include higher number cases here as well, but just to clarify, are you also asking for a test which raises several thousand exceptions to check that the Sentry alert is only called a few times after the initial 100? I feel like that behavior is captured by the test range supplied here, but let me know if you feel that a test which goes through those exceptions is more what you were thinking.